APIs/APIManager.m

```diff
@@ -107,9 +113,49 @@ - (void)getAllGMSPlaces:(void(^)(NSArray<GMSPlace*>* places))completion {
     }];
 }
 
+- (void)getNextGMSPlacesBatch:(void(^)(NSArray<GMSPlace *> *places))completion {
+    PFQuery *query = [PFQuery queryWithClassName:@"Place"];
+    query.limit = kQuerySize;
+    [query orderByDescending:@"createdAt"];
+    if (self.lastQueryDate != nil) {
+        [query whereKey:@"createdAt" lessThan:self.lastQueryDate];
+    }
+    
+    [query findObjectsInBackgroundWithBlock:^(NSArray * _Nullable objects, NSError * _Nullable error) {
+        Place *lastPlace = [objects lastObject];
+        self.lastQueryDate = lastPlace.createdAt;
+        if (error == nil && objects != nil) {
+            NSArray<Place *> *orderedObjects = [NSArray arrayWithArray:objects];
+            
+            Place *placeHolder = [[Place alloc] init];
+            NSMutableArray *places = [NSMutableArray arrayWithCapacity:kQuerySize];
+            for (int i = 0; i < orderedObjects.count; i++) {
+                [places addObject:placeHolder];
+            }
+            
+            __block NSNumber *count;
+            count = [NSNumber numberWithInteger:0];
```

Comment:
I believe you can do __block int count = 0;

using an int will just be a little simpler than an NSNumber as you don't have to keep reconstructing it and getting its intvalue